src/eos-updater-fetch.c

```diff
@@ -43,6 +43,14 @@ content_fetch_finished (GObject *object,
 
   if (error)
     {
+      if (g_error_matches (error, G_IO_ERROR, G_IO_ERROR_FAILED))
+        {
+          g_autofree gchar *old_message = error->message;
```

Comment:
You want to either steal it or duplicate it, otherwise `old_message` will become a dangling pointer after the next line. And you should get a crash after the double free.